File: api/app/schemas/technicians.py
from __future__ import annotations

from pydantic import BaseModel


class TechnicianCreate(BaseModel):
    full_name: str
    username: str
    email: str
    password: str
    phone: str | None = None


class TechnicianProfile(BaseModel):
    id: str
    username: str
    full_name: str
    email: str
    phone: str | None
    station_ids: list[str]
    is_active: bool
    created_at: str

File: web/src/types/technicians.ts
export interface TechnicianProfile {
  id: string;
  username: string;
  full_name: string;
  email: string;
  phone: string | null;
  station_ids: string[];
  is_active: boolean;
  created_at: string;
}

export interface TechnicianCreate {
  full_name: string;
  username: string;
  email: string;
  password: string;
  phone?: string;
}
